### ISSUE TITLE
Chained Nested Children Bindings

### ISSUE THREAD

**Issue by kudla** [pinojs/pino#1627]
Hi
Please is it possible somehow chaining nested children paths in bindings in standard way or at least with simple workaround?
E.g.
```js
const appLogger = pino({module: 'app'})
appLogger.debug('hi')

const moduleLogger = appLogger.child({module: 'module'})
moduleLogger.debug('there')

const submoduleLogger = moduleLogger.child({module: 'sub-module'})
submoduleLogger.debug('world')
```

```sh
2017-07-05T16:15:55.167-0700 [DEBUG] app: hi
2017-07-05T16:15:55.167-0700 [DEBUG] app.module: there
2017-07-05T16:15:55.167-0700 [DEBUG] app.module.sub-module: world
```

Thx

**Comment by jsumners** [comment]
No, this is not supported. Each child adds whatever bindings are supplied, but does not mutate the bindings of any ancestors.

**Comment by marcbachmann** [comment]
That one is quite similar https://github.com/pinojs/pino/issues/416
Maybe you could do something using the `onChild` hook. https://getpino.io/#/docs/api?id=onchild-function

Or wrap it with a custom function that keeps track of the `module` and then calls `.child`

**Comment by kudla** [comment]
Thanks guys
@marcbachmann I had the same ideas on using `onChild` yet it seemed more like post fact notifier to me and not that good place for handling the composition. Besides as I can see there's another mutation point `setBindings`  which wouldn't follow the logic anyway.

I ended up with local specific wrapper extending a proto with keys merging customization. But an Idea of namespaces in #416 looks more appropriate. is it still an actual evolution direction
